### ISSUE TITLE
Defining a ruleset

### ISSUE THREAD

**Issue by prafulbagai** [jruizgit/rules#142]
I've an API endpoint, `/api/create/`. This creates a new entry in the DB depending upon the payload received from the client.

    Payload - { 
    'field1': 'value1`,
    'field2': 'value2`,
    'field3': 'value3`,
    'field4': 'value4`,
    'type': 'A'
    }

Now depending on the `type`, I've a specific ruleset/workflow. For eg:- `Type A` needs to perform certain tasks before saving the data, `Type B` has its own set of operations. 

I don't have any UI, just the requests coming over the POST request. How can I use rules for such a use-case?

**Comment by jruizgit** [comment]
Yes, you can. You can write a rule to start a ruleset or you can have a top level branch rule:

```python
with statechart('risk3'):
    with state('start'):
        @when_all(m.type == 'A')
        @to('A')
        def action_A(c):

        @when_all(m.type == 'B')
        @to('B')
        def action_B(c):
            
    with state('A'):
        # do your thing

    with state('B'):
        # do your thing

run_all()
```

Please take a look at the docs for all the details:

https://github.com/jruizgit/rules/blob/master/docs/py/reference.md
